content/cli/commands/table-format.md

```diff
@@ -0,0 +1,286 @@
+                type: string
+            primaryKey:
+                type: boolean 
+```
+
+#### Field descriptions
+##### name
+Required
+
+This is the name of your Table Format. You'll see this name when you run `fluvio tableformat list`, and you'll use this name with `fluvio consume topic-name --tableformat <name>`
+
+##### inputFormat
+Required
+
+The only supported option for this field is `"JSON"`
+
+##### columns
+optional array - The default column display will be the top-level keys (ordered alphabetically).
+
+Each element references a key from input json object.
+
+The ordering of each element is important, as it will be the order columns will be rendered.
+
+##### keyPath
+This is the only required column field. This should be a top-level key. If the key path doesn't exist, the column will print with no data.
+
+##### headerLabel
+optional - default uses key name. Override the label of the column.
+
+##### primaryKey
+optional - default false. If specified to true, rendering updates to the table will compare the values of primary keys to define a set. When new data matches an existing set, it's row will be updated. Otherwise it will append a new row to the table.
+
+#### Examples
+
+For the following examples, we'll start off with our topic data arriving in this order.
+
+```json
+{"key1":"a","key2":"1","key3":"Alice","id":123}
+{"key1":"b","key2":"2","key3":"Bob","id":456}
+{"key1":"c","key2":"3","key3":"Carol","id":789}
+{"key1":"x","key2":"10","key3":"Alice","id":123}
+{"key1":"y","key2":"20","key3":"Bob","id":456}
+{"key1":"c","key2":"30","key3":"Carol","id":789}
+```
+
+##### Example 0
+
+**No tableformat**
+
+Using the [`full_table`] output without using a tableformat print each key into a column in alphabetical order from left to right.
+
+%copy%
+```shell
+fluvio consume event-data -B --output full_table
+```
+
+Output:
+
+```
+┌('c' to clear table | 'q' or ESC to exit) | Items: 6─────────────────┐
+│id                key1              key2              key3           │
+│123               a                 1                 Alice          │
+│456               b                 2                 Bob            │
+│789               c                 3                 Carol          │
+│123               x                 10                Alice          │
+│456               y                 20                Bob            │
+│789               c                 30                Carol          │
+└─────────────────────────────────────────────────────────────────────┘
+```
+
+
+##### Example 1
+
+**Display a subset of data**
+
+In this example, we only want to display data for only 2 of the keys. The ordering of the columns will be `key1` first, then `key2`.
+
+Config:
+
+%copy%
+```yaml
+name: "exampleformat1"
+inputFormat: "JSON"
+columns:
+  - keyPath: "key1"
+  - keyPath: "key2"
+```
+
+Command:
+
+%copy%
+```shell
+fluvio consume event-data -B --output full_table --tableformat exampleformat1
+```
+
```

Comment:
The command hangs for me:

```
fluvio consume test -B --output full_table --tableformat exampleformat1 
```

Reply:
FYI...  this command works

```
fluvio consume test -B --output full_table                             
Consuming records from the beginning of topic 'test'
```

Reply:
What was missing were steps for creating the TableFormat from the config file. I placed a filename in the config comment, and a command that uses the config. I hope this makes the usage a little more clear.